gr-utils/python/blocktool/tests/test_blocktool.py

```diff
@@ -0,0 +1,187 @@
+    SKIP_BLOCK_TEST = False
+except:
+    SKIP_BLOCK_TEST = True
+
+try:
+    import apt
+    CACHE = apt.cache.Cache()
+    CACHE.open()
+
+    PKG = CACHE['castxml']
+    if PKG.is_installed:
+        SKIP_BLOCK_TEST = False
+    else:
+        SKIP_BLOCK_TEST = True
+except:
+    SKIP_BLOCK_TEST = True
+
+from jsonschema import validate
+
+from blocktool.core.parseheader import BlockHeaderParser
+from blocktool.core.base import BlockToolException
+from blocktool.core import Constants
+from blocktool.core.outputschema import PARSED_OUTPUT_SCHEME
+
+
+class TestBlocktoolCore(unittest.TestCase):
+    """ The Tests for blocktool core """
+
+    def __init__(self, *args, **kwargs):
+        super(TestBlocktoolCore, self).__init__(*args, **kwargs)
+        self.current_folder = os.path.abspath(os.path.dirname(__file__))
+        self.test_dir = os.path.abspath(os.path.join(
+            self.current_folder, '../../../../gr-analog/include/gnuradio/analog'))
+        self.target_file = os.path.abspath(
+            os.path.join(self.test_dir, 'agc2_cc.h'))
+        self.impl_dir = os.path.abspath(os.path.join(self.current_folder,
+                                                     '../../../../gr-analog/lib'))
+        self.modname = os.path.basename(os.path.dirname(
+            os.path.dirname(os.path.dirname(os.path.dirname(self.target_file)))))
+        self.gr = self.modname.split('-')[0]
+        self.module = self.modname.split('-')[-1]
+
+    def is_int(self, number):
+        """
+        Check for int conversion
+        """
+        try:
+            int(number)
+            return True
+        except ValueError:
+            return False
+
+    @unittest.skipIf(SKIP_BLOCK_TEST, 'pygccxml not working, skipping whole unittest')
+    def test_block_parser(self):
+        """
+        Tests for the block parser API
+        """
+        try:
+            warnings.simplefilter("ignore", ResourceWarning)
+        except NameError:
+            pass
+
+        # test for all the Blocktool Exceptions
+        test_dict = {}
+        # test for none type file_path
+        test_dict['file_path'] = None
+        with self.assertRaises(TypeError):
+            BlockHeaderParser(**test_dict).run
+        # test for non-existent header
+        test_dict['file_path'] = os.path.abspath(
+            os.path.join(self.test_dir, 'sample.h'))
+        with self.assertRaises(OSError):
+            BlockHeaderParser(**test_dict).run
+        # test for valid header file
+        test_dict['file_path'] = os.path.abspath(
+            os.path.join(self.test_dir, 'CMakeLists.txt'))
+        if not os.path.basename(test_dict['file_path']).endswith('.h'):
+            with self.assertRaises(BlockToolException):
+                BlockHeaderParser(**test_dict).run
+        # test for invalid header module
+        test_dict['file_path'] = os.path.abspath(
+            os.path.join(self.current_folder, '../../../../gr-utils/python/blocktool/__init__.py'))
```

Comment:
???

Reply:
This was a debug statement. Sorry for the mess. I have now cleaned up the tests.